### PR TITLE
sql/postgres: fixed inspection of multicolumn indexes

### PR DESCRIPTION
**Pull request by svstanev**
I have an unique index like this:

```sql
create unique index if not exists u_entity_name on entity (
  entity_type,
  coalesce(parent_id,0),
  name
);
```

which when inspected become like this:

```hcl
index "u_entity_name" {
    unique = true
    type   = BTREE
    on {
      expr = "COALESCE(parent_id, (0)::bigint)"
    }
    on {
      expr = "COALESCE(parent_id, (0)::bigint)"
    }
    on {
      expr = "COALESCE(parent_id, (0)::bigint)"
    }
  }
```

Expected:
```hcl
index "u_entity_name" {
    unique = true
    type   = BTREE
    on {
      column = column.entity_type
    }
    on {
      expr = "COALESCE(parent_id, (0)::bigint)"
    }
    on {
      column = column.name
    }
  }
```

Reason: incorrect index metadata query.

### CODE REVIEW

File: sql/postgres/inspect.go [hunk 1]
```diff
@@ -387,18 +387,17 @@ func (i *inspect) addIndexes(s *schema.Schema, rows *sql.Rows) error {
 				NullsLast:  nullslast.Bool,
 			})
 		}
-		switch {
-		case sqlx.ValidString(expr):
-			part.X = &schema.RawExpr{
-				X: expr.String,
-			}
-		case sqlx.ValidString(column):
```
Comment (a8m) [inline sql/postgres/inspect.go:395]: Why this change is needed?

File: sql/postgres/inspect.go [hunk 2]
```diff
@@ -777,25 +776,31 @@ SELECT
...
 	obj_description(to_regclass($1 || i.relname)::oid) AS comment
 FROM
-	pg_index idx
+	(
+		select
+			*,
+			generate_series(1,array_length(i.indkey,1)) as ord,
+			unnest(i.indkey) AS key
+		from pg_index i
+	) idx
```
Comment (a8m) [inline sql/postgres/inspect.go:791]: 💯

File: sql/postgres/inspect.go (hunk 1 above)
Comment (svstanev) [reply sql/postgres/inspect.go:395]: Each row of the indexesQuery returns an index part/column in a multicolumn index. As per the suggested query each part has a valid expression but column only if it is created with a column name:

```sql
LEFT JOIN pg_attribute a ON (a.attrelid, a.attnum) = (idx.indrelid, idx.key)
```

Comment (a8m) [reply sql/postgres/inspect.go:395]: Thanks for the explanation. If you don't mind, please use revert to the switch statement and I'll merge it.

**Review by a8m — APPROVED** [review]
💯

**Comment by a8m** [comment]
Thanks for the contribution @svstanev 🙏🏻
